assets/app/scripts/services/data.js

```diff
@@ -641,6 +640,9 @@ angular.module('openshiftConsole')
         return;
       }
 
+      // Reset the retries count once we've received a message with no error.
+      this._watchWebsocketRetries(type, context, 0);
```

Comment:
to test, you can set the read/write timeout artificially short in the api server